server/app/repository/ProgramRepository.java

```diff
@@ -247,23 +249,20 @@ public PaginationResult<Application> getApplicationsForAllProgramVersions(
     }
 
     if (searchNameFragment.isPresent() && !searchNameFragment.get().isBlank()) {
-      String search = searchNameFragment.get();
+      String search = searchNameFragment.get().trim();
 
       if (search.matches("^\\d+$")) {
         query = query.eq("id", Integer.parseInt(search));
       } else {
-        search = search.toLowerCase(java.util.Locale.ROOT);
-
+        String firstNamePath = getApplicationObjectPath(WellKnownPaths.APPLICANT_FIRST_NAME);
+        String lastNamePath = getApplicationObjectPath(WellKnownPaths.APPLICANT_LAST_NAME);
         query =
             query
                 .or()
-                .eq("submitter_email", search)
-                .raw(
-                    getApplicationObjectPath("applicant", "name", "first_name") + " ILIKE ?",
-                    "%" + search + "%")
-                .raw(
-                    getApplicationObjectPath("applicant", "name", "last_name") + " ILIKE ?",
-                    "%" + search + "%")
+                .ieq("submitter_email", search)
+                .raw(firstNamePath + " || ' ' || " + lastNamePath + " ILIKE ?", "%" + search + "%")
```

Comment:
Note to the reviewer: JSONB allows for concatenated strings to be passed to ILIKE, similar to standard string columns: https://www.postgresql.org/docs/9.5/functions-json.html